packages/wdio-config/tests/configparser.test.ts

```diff
@@ -728,39 +728,55 @@ describe('ConfigParser', () => {
             const configParser = await ConfigParserForTest()
             await configParser.addConfigFile(FIXTURES_CONF)
             configParser.merge({ spec: [INDEX_PATH, FIXTURES_CONF] })
-            configParser.merge({ exclude: [path.resolve(__dirname).replace(/\\/g, '/') + '/*'] })
-            const specs = configParser.getSpecs()
-            expect(specs).toHaveLength(2)
+
+            // first validate that the conf fixture 'spec' was merged successfully
+            expect(configParser.getSpecs()).toHaveLength(1)
+
+            configParser.merge({ exclude: [path.join(__dirname, '**', '*conf*')] })
+
+            // then after merging an exclude containing a glob pattern, validate that the exclude
+            // attribute contains multiple items and the filtering on the spec attribute works
+            expect(configParser.getConfig().exclude.length).toBeGreaterThan(0)
+            expect(configParser.getSpecs()).toHaveLength(0)
         })
 
-        it('should overwrite exclude if piped into cli command', async () => {
+        it('should overwrite config exclude if piped into cli command', async () => {
             const configParser = await ConfigParserForTestWithAllFiles()
             await configParser.addConfigFile(FIXTURES_CONF)
+            // check that the initial exclude from the config filtered out 1 spec (out of a total of 4)
+            expect(configParser.getSpecs()).toHaveLength(3)
+
             configParser.merge({ exclude: [INDEX_PATH] })
-            const specs = configParser.getSpecs()
-            expect(specs).toHaveLength(4)
+            // after overriding the config exclude with cli exclude check that the initial config exclude is discarded
+            expect(configParser.getSpecs()).toHaveLength(4)
         })
 
-        it('should overwrite exclude if piped into cli command with params', async () => {
+        it('should overwrite config and capabilities exclude if piped into cli command', async () => {
             const configParser = await ConfigParserForTestWithAllFiles()
             await configParser.addConfigFile(FIXTURES_CONF)
-            configParser.merge({})
+            expect(configParser.getSpecs()).toHaveLength(3)
 
-            const utilsPath = path.join(__dirname, '..', 'src', 'utils.ts')
-            const indexPath = path.join(__dirname, '..', 'src', 'index.ts')
-            const specs = configParser.getSpecs([indexPath, utilsPath], [utilsPath])
-            expect(specs).toEqual([indexPath])
+            configParser.merge({ exclude: [FIXTURES_CONF] })
+            const specs = configParser.getSpecs([FIXTURES_CONF, FIXTURES_CONF_RDC], [FIXTURES_CONF_RDC])
+            expect(specs).toEqual([FIXTURES_CONF_RDC])
```

Comment:
there were multiple problems with this test:
1. no CLI exclude param was provided
2. because of the [issue mentioned before](https://github.com/webdriverio/webdriverio/pull/9013/files#r1002671129), when taking into account the validation:
```
            const specs = configParser.getSpecs([indexPath, utilsPath], [utilsPath])
            expect(specs).toEqual([indexPath])
```
even if a bug were introduced which made it such that the `utilsPath` capability exclude is no longer taken into account when it should, the above statement would still pass, missing the bug